configure.ac

```diff
@@ -1475,7 +1475,22 @@ AC_ARG_ENABLE(omkafka,
          esac],
         [enable_omkafka=no]
 )
+AC_ARG_ENABLE(kafka_tests,
+        [AS_HELP_STRING([--enable-kafka-tests],[enable Kafka specific tests in testbench @<:@default=no@:>@])],
+        [case "${enableval}" in
+         yes) enable_kafka_tests="yes" ;;
+          no) enable_kafka_tests="no" ;;
+           *) AC_MSG_ERROR(bad value ${enableval} for --enable-kafka-tests) ;;
+         esac],
+        [enable_kafka_tests=no]
+)
+AM_CONDITIONAL(ENABLE_KAFKA_TESTS, test x$enable_kafka_tests = xyes)
+
 if test "x$enable_omkafka" = "xyes"; then
+	if test "x$enable_kafka_tests" = "xyes"; then
+		AX_PROG_JAVAC #we don't need javac, but macro documentation says JAVAC *must* be checked before JAVA
+		AX_PROG_JAVA
+	fi
```

Comment:
2016-04-19 18:02 GMT+02:00 Janmejay Singh <notifications@github.com>:

> In configure.ac
> <https://github.com/rsyslog/rsyslog/pull/967#discussion_r60259790>:
>
> >  if test "x$enable_omkafka" = "xyes"; then
> > +	if test "x$enable_kafka_tests" = "xyes"; then
> > +		AX_PROG_JAVAC #we don't need javac, but macro documentation says JAVAC *must* be checked before JAVA
> > +		AX_PROG_JAVA
> > +	fi
>
> This will run only if the user enables kafka-tests. It is disabled by
> default, so it won't affect anyone unless they enable kafka-tests.
>

sorry, I think I should call it a day. I have overlooked the second if
condition...

Rainer

> If they enable it, this will fail in an expected way which will pin-point
> the cause of failure.
>
> I felt its better to fail upfront (people expect to catch
> incompatibilities in configure, right?).
>
> The test silently ignoring it will make it more hidden, isn't it?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/rsyslog/rsyslog/pull/967/files/003b8a80ff4c37a4c1b93e34a670c225980b170d#r60259790>
>